File: src/main/java/seremis/geninfusion/soul/traits/Trait.java
package seremis.geninfusion.soul.traits;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.entity.IEntityLivingData;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.util.ChunkCoordinates;
import net.minecraft.util.DamageSource;
import seremis.geninfusion.api.soul.IEntitySoulCustom;
import seremis.geninfusion.api.soul.ITrait;

public class Trait implements ITrait {

    @Override
    public void onUpdate(IEntitySoulCustom entity) {}

    @Override
    public boolean interact(IEntitySoulCustom entity, EntityPlayer player) {
        return true;
    }

    @Override
    public void onDeath(IEntitySoulCustom entity, DamageSource source) {}

    @Override
    public void onKillEntity(IEntitySoulCustom entity, EntityLivingBase killed) {}

    @Override
    public boolean attackEntityFrom(IEntitySoulCustom entity, DamageSource source, float damage) {
        return true;
    }

    @Override
    public void onSpawnWithEgg(IEntitySoulCustom entity, IEntityLivingData data) {}

    @Override
    public void playSound(IEntitySoulCustom entity, String name, float volume, float pitch) {}

    @Override
    public void damageEntity(IEntitySoulCustom entity, DamageSource source, float damage) {}

    @Override
    public void updateAITick(IEntitySoulCustom entity) {}

    @Override
    public void firstTick(IEntitySoulCustom entity) {}

    @Override
    public boolean attackEntityAsMob(IEntitySoulCustom entity, Entity entityToAttack) {
        return true;
    }

    @Override
    public void attackEntity(IEntitySoulCustom entity, Entity entityToAttack, float distance) {}

    @Override
    public Entity findPlayerToAttack(IEntitySoulCustom entity) {
        return null;
    }

    @Override
    public float applyArmorCalculations(IEntitySoulCustom entity, DamageSource source, float damage) {
        return 0.0F;
    }

    @Override
    public float applyPotionDamageCalculations(IEntitySoulCustom entity, DamageSource source, float damage) {
        return 0.0F;
    }

    @Override
    public void damageArmor(IEntitySoulCustom entity, float damage) {}

    @Override
    public void setOnFireFromLava(IEntitySoulCustom entity) {}

    @Override
    public float getBlockPathWeight(IEntitySoulCustom entity, int x, int y, int z) {
        return 0.0F;
    }

    @Override
    public void updateEntityActionState(IEntitySoulCustom entity) {}

    @Override
    public void updateWanderPath(IEntitySoulCustom entity) {}

    @Override
    public void attackEntityWithRangedAttack(IEntitySoulCustom entity, EntityLivingBase target, float distanceModified) {}

    @Override
    @SideOnly(Side.CLIENT)
    public void render(IEntitySoulCustom entity, float timeModifier, float walkSpeed, float specialRotation, float rotationYawHead, float rotationPitch, float scale) {}

    @Override
    public boolean isWithinHomeDistanceCurrentPosition(IEntitySoulCustom entity) {
        return false;
    }

    @Override
    public boolean isWithinHomeDistance(IEntitySoulCustom entity, int x, int y, int z) {
        return false;
    }

    @Override
    public ChunkCoordinates getHomePosition(IEntitySoulCustom entity) {
        return null;
    }

    @Override
    public void setHomeArea(IEntitySoulCustom entity, int x, int y, int z, int maxDistance) {}

    @Override
    public float getMaxHomeDistance(IEntitySoulCustom entity) {
        return 0.0F;
    }

    @Override
    public void detachHome(IEntitySoulCustom entity) {}

    @Override
    public boolean hasHome(IEntitySoulCustom entity) {
        return false;
    }

    @Override
    public void writeToNBT(IEntitySoulCustom entity, NBTTagCompound compound) {}

    @Override
    public void readFromNBT(IEntitySoulCustom entity, NBTTagCompound compound) {}

    @Override
    public String getEntityTexture(IEntitySoulCustom entity) {
        return null;
    }
}

File: src/main/java/seremis/geninfusion/soul/traits/Trait.scala
package seremis.geninfusion.soul.traits

import cpw.mods.fml.relauncher.{Side, SideOnly}
import net.minecraft.entity.effect.EntityLightningBolt
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.entity.{Entity, EntityLivingBase, IEntityLivingData}
import net.minecraft.nbt.NBTTagCompound
import net.minecraft.util.{ChunkCoordinates, DamageSource}
import seremis.geninfusion.api.soul.{IEntitySoulCustom, ITrait}

class Trait extends ITrait {

    override def onUpdate(entity: IEntitySoulCustom) {}

    override def interact(entity: IEntitySoulCustom, player: EntityPlayer): Boolean = false

    override def onDeath(entity: IEntitySoulCustom, source: DamageSource) {}

    override def onKillEntity(entity: IEntitySoulCustom, killed: EntityLivingBase) {}

    override def attackEntityFrom(entity: IEntitySoulCustom, source: DamageSource, damage: Float): Boolean = false

    override def onSpawnWithEgg(entity: IEntitySoulCustom, data: IEntityLivingData) {}

    override def playSound(entity: IEntitySoulCustom, name: String, volume: Float, pitch: Float) {}

    override def damageEntity(entity: IEntitySoulCustom, source: DamageSource, damage: Float) {}

    override def updateAITick(entity: IEntitySoulCustom) {}

    override def firstTick(entity: IEntitySoulCustom) {}

    override def attackEntityAsMob(entity: IEntitySoulCustom, entityToAttack: Entity): Boolean = false

    override def attackEntity(entity: IEntitySoulCustom, entityToAttack: Entity, distance: Float) {}

    override def findPlayerToAttack(entity: IEntitySoulCustom): Entity = null

    override def applyArmorCalculations(entity: IEntitySoulCustom, source: DamageSource, damage: Float): Float = 0.0F

    override def applyPotionDamageCalculations(entity: IEntitySoulCustom, source: DamageSource, damage: Float): Float = 0.0F

    override def damageArmor(entity: IEntitySoulCustom, damage: Float) {}

    override def setOnFireFromLava(entity: IEntitySoulCustom) {}

    override def getBlockPathWeight(entity: IEntitySoulCustom, x: Int, y: Int, z: Int): Float = 0.0F

    override def updateEntityActionState(entity: IEntitySoulCustom) {}

    override def updateWanderPath(entity: IEntitySoulCustom) {}

    override def attackEntityWithRangedAttack(entity: IEntitySoulCustom, target: EntityLivingBase, distanceModified: Float) {}

    override def isWithinHomeDistanceCurrentPosition(entity: IEntitySoulCustom): Boolean = false

    override def isWithinHomeDistance(entity: IEntitySoulCustom, x: Int, y: Int, z: Int): Boolean = false

    override def getHomePosition(entity: IEntitySoulCustom): ChunkCoordinates = null

    override def setHomeArea(entity: IEntitySoulCustom, x: Int, y: Int, z: Int, maxDistance: Int) {}

    override def getMaxHomeDistance(entity: IEntitySoulCustom): Float = 0.0F

    override def detachHome(entity: IEntitySoulCustom) {}

    override def hasHome(entity: IEntitySoulCustom): Boolean = false

    override def writeToNBT(entity: IEntitySoulCustom, compound: NBTTagCompound) {}

    override def readFromNBT(entity: IEntitySoulCustom, compound: NBTTagCompound) {}

    override def onStruckByLightning(entity: IEntitySoulCustom, lightingBolt: EntityLightningBolt) {}

    @SideOnly(Side.CLIENT)
    override def render(entity: IEntitySoulCustom, timeModifier: Float, limbSwing: Float, specialRotation: Float, rotationYawHead: Float, rotationPitch: Float, scale: Float) {}

    @SideOnly(Side.CLIENT)
    override def preRenderCallback(entity: IEntitySoulCustom, partialTickTime: Float) {}

    @SideOnly(Side.CLIENT)
    override def inheritRenderPass(entity: IEntitySoulCustom, renderPass: Int, partialTickTime: Float): Int = 0

    @SideOnly(Side.CLIENT)
    override def shouldRenderPass(entity: IEntitySoulCustom, renderPass: Int, partialTickTime: Float): Int = 0

    @SideOnly(Side.CLIENT)
    override def getColorMultiplier(entity: IEntitySoulCustom, brightness: Float, partialTickTime: Float): Int = 0

    @SideOnly(Side.CLIENT)
    override def renderEquippedItems(entity: IEntitySoulCustom, partialTickTime: Float) {}

    @SideOnly(Side.CLIENT)
    override def getEntityTexture(entity: IEntitySoulCustom): String = null

    override def setDead(entity: IEntitySoulCustom) {}

    override def setCustomNameTag(entity: IEntitySoulCustom, nameTag: String) {}

    override def getCustomNameTag(entity: IEntitySoulCustom): String = null

    override def hasCustomNameTag(entity: IEntitySoulCustom): Boolean = false
}